PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -857,8 +877,8 @@ LoadBootImages (
     ContainerImage = &BootImage[Index].ContainerImage;
     if ((ContainerImage->Indicate == '!') && (ContainerImage->BackSlash == '/')) {
       Status = GetBootImageFromIfwiContainer (OsBootOption, LoadedImage);
-    } else if (FsHandle != NULL) {
-      Status = GetBootImageFromFs (FsHandle, OsBootOption, LoadedImage);
+    } else if (BootFromFs) {
```

Comment:
Maybe we could use filename as below to know if it is boot from file system. 
} else if ((AsciiCharToUpper(OsBootOption->Image[Index].FileImage.FileName[0]) >= 'A') && AsciiCharToUpper(OsBootOption->Image[Index].FileImage.FileName[0]) <= 'Z') {

Reply:
Here checking `OsBootOption->Image[Index].LbaImage.Valid == 1` might be better. `\1` cannot be the first char in filename.